flow/connectors/core.go

```diff
@@ -190,6 +191,39 @@ func GetQRepSyncConnector(ctx context.Context, config *protos.Peer) (QRepSyncCon
 	}
 }
 
+func GetConnector(ctx context.Context, config *protos.Peer) (Connector, error) {
+	inner := config.Type
+	switch inner {
+	case protos.DBType_POSTGRES:
+		pgConfig := config.GetPostgresConfig()
+		if pgConfig == nil {
+			return nil, ErrWrongConfig
+		}
+		return connpostgres.NewPostgresConnector(ctx, pgConfig)
+	case protos.DBType_BIGQUERY:
+		bqConfig := config.GetBigqueryConfig()
+		if bqConfig == nil {
+			return nil, ErrWrongConfig
+		}
+		return connbigquery.NewBigQueryConnector(ctx, bqConfig)
+
+	case protos.DBType_SNOWFLAKE:
+		sfConfig := config.GetSnowflakeConfig()
+		if sfConfig == nil {
+			return nil, ErrWrongConfig
+		}
+		return connsnowflake.NewSnowflakeConnector(ctx, sfConfig)
+	// case protos.DBType_S3:
+	// 	return conns3.NewS3Connector(ctx, config.GetS3Config())
+	// case protos.DBType_EVENTHUB:
+	// 	return connsqlserver.NewSQLServerConnector(ctx, config.GetSqlserverConfig())
+	// case protos.DBType_SQLSERVER:
+	// 	return conneventhub.NewEventHubConnector(ctx, config.GetEventhubConfig())
+	default:
+		return nil, ErrUnsupportedFunctionality
```

Comment:
can you return a more detailed error with the info of the Peer and that it isn't supported?